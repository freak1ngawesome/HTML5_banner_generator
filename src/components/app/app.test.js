import React from 'react';
import ReactDOM from 'react-dom';
import App from './app';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import store from '../../store';

describe('Testing <App/> rendering without crush', () => {
  const app = renderer.create(
  <Provider store={store}>
    <App/>
  </Provider>)
  .toJSON();

  it('Component has rendered correctly',() => {
    expect(app).toMatchSnapshot();
  });

  it('No crush, please...',() => {
    const div = document.createElement('div');
    ReactDOM.render(
    <Provider store={store}>
      <App/>
    </Provider>, div);
  });

});