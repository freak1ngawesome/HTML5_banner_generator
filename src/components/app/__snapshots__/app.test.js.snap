// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <App/> rendering without crush Component has rendered correctly 1`] = `
<div
  className="container"
>
  <header
    className="header"
  >
    <h1
      className="header__heading"
    >
      Create your awesome banner
    </h1>
  </header>
  <section
    className="export"
  >
    <button
      className="export__btn save_png"
      onClick={[Function]}
    >
      Save as .png
    </button>
    <button
      className="export__btn copy_html"
      onClick={[Function]}
    >
      Copy as HTML5 string
    </button>
    <button
      className="export__btn copy_json"
      onClick={[Function]}
    >
      Copy settings
    </button>
  </section>
  <p
    className="notification"
  />
  <section
    className="customization"
  >
    <div
      className="customization__block"
    >
      <button
        className="customizer_btn"
        onClick={[Function]}
      >
        show
      </button>
      <form
        className="customization__form"
      >
        <div
          className="customization__form_item banner_sizes"
        >
          <h2
            className="customization__form_label"
          >
            Banner size
          </h2>
          <div
            className="customization__form_inputs"
          >
            <span>
              X:
            </span>
            <input
              className="customization__form_text"
              onChange={[Function]}
              type="text"
              value="282"
            />
            <span>
              Y:
            </span>
            <input
              className="customization__form_text"
              onChange={[Function]}
              type="text"
              value="376"
            />
          </div>
        </div>
        <div
          className="customization__form_item background"
        >
          <h2
            className="customization__form_label"
          >
            Background
          </h2>
          <div
            className="customization__form_inputs"
          >
            <input
              className="customization__form_color"
              onInput={[Function]}
              type="color"
              value="#000000"
            />
            <input
              className="customization__form_text"
              onChange={[Function]}
              type="text"
              value="#000000"
            />
          </div>
        </div>
        <div
          className="customization__form_item font_color"
        >
          <h2
            className="customization__form_label"
          >
            Font color
          </h2>
          <div
            className="customization__form_inputs"
          >
            <input
              className="customization__form_color"
              onInput={[Function]}
              type="color"
              value="#ffffff"
            />
            <input
              className="customization__form_text"
              onChange={[Function]}
              type="text"
              value="#ffffff"
            />
          </div>
        </div>
        <div
          className="customization__form_item font_size"
        >
          <h2
            className="customization__form_label"
          >
            Font size
          </h2>
          <div
            className="customization__form_inputs"
          >
            <input
              className="customization__form_text"
              onChange={[Function]}
              type="text"
              value="44"
            />
          </div>
        </div>
        <div
          className="customization__form_item imgURL"
        >
          <h2
            className="customization__form_label"
          >
            Enter img link or URI
          </h2>
          <div
            className="customization__form_inputs"
          >
            <input
              className="customization__form_text"
              onChange={[Function]}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="customization__form_item link"
        >
          <h2
            className="customization__form_label"
          >
            Enter link to your site
          </h2>
          <div
            className="customization__form_inputs"
          >
            <input
              className="customization__form_text"
              onChange={[Function]}
              type="text"
              value=""
            />
          </div>
        </div>
        <button
          className="reset_btn"
          onClick={[Function]}
        >
          reset
        </button>
      </form>
    </div>
  </section>
  <section
    className="preview"
    id="preview"
  >
    <div
      className="banner"
      id="capture"
      style={
        Object {
          "background": "#000000",
          "borderRadius": "10px",
          "height": "376px",
          "overflow": "hidden",
          "position": "absolute",
          "width": "282px",
        }
      }
    >
      <img
        alt=""
        id="banner_img"
        src=""
        style={
          Object {
            "left": "0",
            "position": "absolute",
            "top": "0",
          }
        }
      />
      <div
        id="banner_text"
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "monospace",
            "fontSize": "44px",
            "fontWeight": "700",
            "height": "132px",
            "left": "10px",
            "lineHeight": "44px",
            "overflow": "hidden",
            "position": "absolute",
            "top": "50%",
            "userSelect": "none",
            "width": "100%",
          }
        }
        suppressContentEditableWarning={true}
      >
         Write your awesome text here
      </div>
    </div>
  </section>
</div>
`;
